projects/rocsparse/library/src/precond/bsrilu0/rocsparse_bsrilu0_kernel_2_8.cpp

```diff
@@ -0,0 +1,544 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * ************************************************************************ */
+
+#include "rocsparse_bsrilu0_kernel_2_8.hpp"
+#include "rocsparse_bsrilu0_info.hpp"
+#include "rocsparse_common.hpp"
+#include "rocsparse_utility.hpp"
+
+namespace rocsparse
+{
+    template <uint32_t BLOCKSIZE,
+              uint32_t WFSIZE,
+              uint32_t BSRDIM,
+              typename T,
+              typename I,
+              typename J>
+    ROCSPARSE_DEVICE_ILF void bsrilu0_device_2_8(rocsparse_direction dir,
+                                                 J                   mb,
+                                                 const I* __restrict__ bsr_row_ptr,
+                                                 const J* __restrict__ bsr_col_ind,
+                                                 T* __restrict__ bsr_val,
+                                                 const I* __restrict__ bsr_diag_ind,
+                                                 J block_dim,
+                                                 int32_t* __restrict__ done_array,
+                                                 const J* __restrict__ map,
+                                                 J* __restrict__ zero_pivot,
+                                                 rocsparse_index_base idx_base,
+                                                 int                  boost,
+                                                 double               boost_tol,
+                                                 T                    boost_val)
+    {
+        // Current row this wavefront is working on
+        J row = map[blockIdx.x];
+
+        // Diagonal entry point of the current row
+        I row_diag = bsr_diag_ind[row];
+
+        // Row entry point
+        I row_begin = bsr_row_ptr[row] - idx_base;
+        I row_end   = bsr_row_ptr[row + 1] - idx_base;
```

Comment:
These variables can be `const`

```suggestion
        const J row = map[blockIdx.x];

        // Diagonal entry point of the current row
        const I row_diag = bsr_diag_ind[row];

        // Row entry point
        const I row_begin = bsr_row_ptr[row] - idx_base;
        const I row_end   = bsr_row_ptr[row + 1] - idx_base;
```

Reply:
100% agree, however, I didn't touch the kernels. This would be changed in a different PR to have a clear picture of the modifications.